code/procs/scanprocs.dm

```diff
@@ -109,10 +110,48 @@
 
 
 			var/bad_stuff = 0
-			if (L.implant && L.implant.len > 0)
+			if (length(L.implant))
+				var/list/implant_list = list()
 				for (var/obj/item/implant/I in L.implant)
 					if (istype(I, /obj/item/implant/projectile))
-						bad_stuff ++
+						bad_stuff++
+						continue
+					if (istype(I, /obj/item/implant/health))
+						if (!istype(I, /obj/item/implant/health/security/anti_mindslave))
+							implant_list["Health"]++
+						else
+							implant_list["Mind-protection"]++
+					else if (istype(I, /obj/item/implant/cloner))
+						implant_list["Cloner"]++
+					else if (istype(I, /obj/item/implant/antirot))
+						implant_list["Rotbusttec"]++
+					else if (istype(I, /obj/item/implant/robotalk))
+						implant_list["Machine translator"]++
+					else if (istype(I, /obj/item/implant/access))
+						implant_list["Access"]++
+					else if (istype(I, /obj/item/implant/tracking))
+						implant_list["Tracking"]++
+					else if (istype(I, /obj/item/implant/robust))
+						implant_list["Robustec"]++
+					else if (istype(I, /obj/item/implant/counterrev))
+						implant_list["Counter-revolutionary"]++
+					else if (syndicate)
+						if (istype(I, /obj/item/implant/revenge/microbomb))
+							implant_list["Microbomb"]++
+						else if (istype(I, /obj/item/implant/revenge/zappy))
+							implant_list["Flyzapper"]++
+						else if (istype(I, /obj/item/implant/mindslave))
+							if (!istype(I, /obj/item/implant/mindslave/super))
+								implant_list["Mindslave"]++
+							else
+								implant_list["Mindslave Deluxe"]++
+						else if (istype(I, /obj/item/implanter/freedom))
+							implant_list["Freedom"]++
```

Comment:
Again, this seems like something that might be better defined on the implants rather than hardcoded into a big elif chain